src/plugins/legacy/medPACS/medPACS.h

```diff
@@ -0,0 +1,301 @@
+                      unsigned int pi_uiLevel, QString parentKey) override;
+    bool createPath(QList<levelMinimalEntries> &pio_path, const datasetAttributes &pi_attributes, unsigned int pi_uiLevel,
+               QString parentKey) override;
+
+    bool alterMetaData(const datasetAttributes &pi_attributes, unsigned int pi_uiLevel, QString key) override;
+
+    bool getThumbnail(QPixmap &po_thumbnail, unsigned int pi_uiLevel, QString key) override;
+
+    bool setThumbnail(QPixmap &pi_thumbnail, unsigned int pi_uiLevel, QString key) override;
+
+    bool commitData(QVariant data, levelMinimalEntries &pio_minimalEntries, unsigned int pi_uiLevel,
+                    QString parentKey) override;
+
+    int push(unsigned int pi_uiLevel, QString key) override;
+
+    QVariant getAsyncResults(int pi_iRequest) override;
+public slots:
+    void abort(int pi_iRequest) override;
+    void onSettingsSaved();
+
+private:
+    void computeDateRange();
+
+    QList<QMap<DcmTagKey, QString>> cFindPatient(const QString &patientID);
+    QList<QMap<DcmTagKey, QString>> cFindStudy(const QString &studyInstanceUID, const QString &patientID = "");
+    QList<QMap<DcmTagKey, QString>> cFindSeries(const QString &studyInstanceUID);
+
+    QList<levelMinimalEntries> getPatientMinimalEntries(const QList<QMap<DcmTagKey, QString>> &infosMap);
+    QList<levelMinimalEntries> getStudyMinimalEntries(const QList<QMap<DcmTagKey, QString>> &infosMap, bool isSeriesFiltered = false);
+    QList<levelMinimalEntries> getSeriesMinimalEntries(const QList<QMap<DcmTagKey, QString>> &infosMap);
+
+    QList<QMap<QString, QString>> getPatientMandatoryAttributes(const QList<QMap<DcmTagKey, QString>> &infosMap);
+    QList<QMap<QString, QString>> getStudyMandatoryAttributes(QList<QMap<DcmTagKey, QString>> &infosMap, bool isPatientFiltered, bool isSeriesFiltered);
+    QList<QMap<QString, QString>> getSeriesMandatoryAttributes(QList<QMap<DcmTagKey, QString>> &infosMap, bool isPatientFiltered, bool isStudiesFiltered);
+
+
+    // Filter parameters
+    const DcmDictEntry * retrieveDcmTag(const QString &key);
+    void computeBirthDateRange(const DcmTagKey &tagKey, const QString &start, const QString &end);
+    void addPatientFilterParameters(const QJsonObject &patientObject);
+    void addStudyFilterParameters(const QJsonObject &studyObject);
+    void addSeriesFilterParameters(const QJsonObject &seriesObject);
+    void readJsonFile(const QString &filePath);
+    void resetFilterParameterValues(QList<medAbstractParameter*> levelFilter, QMap<DcmTagKey, QString> &levelAttributes);
+
+    int getPACSAsyncData(unsigned int pi_uiLevel, const QString &key);
+    int getDicomWebAsyncData(unsigned int pi_uiLevel, const QString &key);
+    
+
+   private:
+    medPACSParametersPrivate* d;
+
+    static std::atomic<int> s_RequestId;
+    static medStringParameter s_Aetitle;
+    static medStringParameter s_Hostname;
```

Comment:
static std::atomic<int> s_RequestId;
    static medStringParameter s_Aetitle;
    static medStringParameter s_Hostname;
    static medIntParameter s_Port;

can not be static